src/grasshopper/lib/fixtures/__init__.py

```diff
@@ -531,3 +564,87 @@ def type_check_list_of_strs(list_of_strs):
             all_strs = all_strs and type(s) == str
         check_passed = all_strs
     return check_passed
+
+
+def _get_composite_weighted_user_classes(
+    full_scenarios_list, composite_scenario_contents
+):
+    """Generate a dictionary of journey classes with their associated weights."""
+    weighted_user_classes = {}
+    child_scenario_specs = _get_child_scenario_specs(
+        full_scenarios_list, composite_scenario_contents
+    )
+    for child_scenario_spec in child_scenario_specs:
+        base_journey_class = _import_class_with_journey(
+            absolute_file_path=f"{os.getcwd()}/"
+            f'{child_scenario_spec.get("test_file_name")}'
+        )
+        base_class_name = base_journey_class.__name__
+        composite_class_name = (
+            f"composite_journey_class_{base_class_name}_{uuid.uuid4()}"
+        )
+
+        # dynamically create a child class which inherits from the base class,
+        # required for having separate scenario args
+        child_journey_class = type(
+            composite_class_name,
+            (base_journey_class,),
+            {
+                "_incoming_test_parameters": child_scenario_spec.get(
+                    "grasshopper_scenario_args"
+                )
+            },
+        )
+        weighted_user_classes[child_journey_class] = child_scenario_spec.get("weight")
+    return weighted_user_classes
```

Comment:
i feel like we might want some debug logging statements through this chunk of code because if it doesn't work, it's all ephemeral and it will be hard to figure out what is going on.

also, for the same reason we might want the generation of the new class in a try/catch block